src/Sql/Sql/Elastic Jobs/Cmdlet/JobExecution/GetAzureSqlElasticJobStepExecution.cs

```diff
@@ -0,0 +1,266 @@
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.ElasticJobs.Cmdlet.JobExecution
+{
+    /// <summary>
+    /// Defines the Get-AzureRmSqlElasticJobExecution Cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlElasticJobStepExecution",
+        DefaultParameterSetName = DefaultParameterSet)]
+    [OutputType(typeof(AzureSqlElasticJobExecutionModel))]
+    public class GetAzureSqlElasticJobStepExecution : AzureSqlElasticJobExecutionCmdletBase<AzureSqlElasticJobExecutionModel>
+    {
+        /// <summary>
+        /// The parameter sets
+        /// </summary>
+        protected const string GetJobStepExecutionDefaultParam = "WithJobStepName";
+        protected const string GetJobStepExecutionJobExecutionModel = "WithJobStepNameUsingParentObject";
+        protected const string GetJobStepExecutionParentResourceId = "WithJobStepNameUsingParentResourceId";
+
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "The resource group name.")]
+        [Parameter(ParameterSetName = GetJobStepExecutionDefaultParam,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the job execution object input model
+        /// </summary>
+        [Parameter(
+            ParameterSetName = InputObjectParameterSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The agent object.")]
+        [Parameter(
+            ParameterSetName = GetJobStepExecutionJobExecutionModel,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The agent object.")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlElasticJobExecutionModel ParentObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the job execution resource id
+        /// </summary>
+        [Parameter(ParameterSetName = ResourceIdParameterSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The job execution resource id.")]
+        [Parameter(ParameterSetName = GetJobStepExecutionParentResourceId,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The job execution resource id.")]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
```

Comment:
@johnpaulkee same comment about `-ResourceId`

Reply:
As discussed we'll address the Resource Id changes in a separate PR to reduce complexity of this one. We can do a one-off release with these changes and then add -ResourceId parameter  later.